### ISSUE TITLE
Binding Generator fails: internal error: entered unreachable code: Unsupported decl for OpenCV namespace: Entity

### ISSUE THREAD

**Issue by Hpmason** [twistedfall/opencv-rust#519]
I'm running into an error while compiling opencv
```
thread 'main' panicked at C:\Users\hpmas\.cargo\registry\src\index.crates.io-6f17d22bba15001f\opencv-binding-generator-0.80.0\src\walker.rs:100:17:
internal error: entered unreachable code: Unsupported decl for OpenCV namespace: Entity {
    kind: LinkageSpec,
    display_name: None,
     location: Some(
         SourceLocation {
             file: Some(
                 File {
                     path: "C:\\dev\\OpenCV\\opencv\\build\\include\\opencv2/core/utility.hpp",
                 },
             ),
             line: 162,
             column: 8,
             offset: 6790,
         },
     ),
 }
```
The line of [utility.hpp is here](https://github.com/opencv/opencv/blob/408730b7aba34bd7256234d3d33d80f623da87a7/modules/core/include/opencv2/core/utility.hpp#L162)
1. Operating system
Windows 11
2. The way you installed OpenCV: package, official binary distribution, manual compilation, etc.
Tried with both vcpkg and official binaries
3. OpenCV version
4.8.0
4. rustc version (`rustc --version`)
rustc 1.74.0 (79e9716c9 2023-11-13)
5. Attach the full output of the following command from your project directory:
   ```shell script
   RUST_BACKTRACE=full cargo build -vv 
   ```
[output.txt](https://github.com/twistedfall/opencv-rust/files/13539716/output.txt)

**Comment by twistedfall** [comment]
Looks like it's clang-17 change that broke it. I'll take a look, meanwhile as a workaround you could try switching back to clang-16

**Comment by Hpmason** [comment]
Switching back to clang-16 seems to work!

**Comment by twistedfall** [comment]
This should now be fixed in 0.88.2
